src/main/kotlin/com/hibob/academy/employee_feedback_feature/dao/FeedbackDao.kt

```diff
@@ -0,0 +1,62 @@
+package com.hibob.academy.employee_feedback_feature.dao
+
+import org.jooq.DSLContext
+import org.jooq.RecordMapper
+import org.jooq.Record
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+@Component
+class FeedbackDao @Autowired constructor(private val sql: DSLContext) {
+    private val feedback = FeedbackTable.instance
+
+    private val feedbackDataMapper = RecordMapper<Record, FeedbackData> { record ->
+        FeedbackData(
+            record[feedback.id],
+            record[feedback.employeeId],
+            record[feedback.companyId],
+            record[feedback.feedbackText],
+            record[feedback.isAnonymous],
+            record[feedback.department],
+            record[feedback.createdAt],
+            record[feedback.status]
+        )
+    }
+
+    fun submitFeedback(newFeedback: FeedbackSubmission): Long {
+        val employeeId = if (newFeedback.isAnonymous) null else newFeedback.employeeId
+
+        return sql.insertInto(feedback)
+            .set(feedback.employeeId, employeeId)
+            .set(feedback.companyId, newFeedback.companyId)
+            .set(feedback.feedbackText, newFeedback.feedbackText)
+            .set(feedback.isAnonymous, newFeedback.isAnonymous)
+            .set(feedback.department, newFeedback.department)
+            .returning(feedback.id)
+            .fetchOne()!!
+            .get(feedback.id)
+    }
+
+    fun getFeedbackByFeedbackId(companyId: Long, feedbackId: Long): FeedbackData? {
+        return selectFeedback()
+            .where(feedback.id.eq(feedbackId))
+            .and(feedback.companyId.eq(companyId))
+            .fetchOne(feedbackDataMapper)
+    }
+
+    fun getFeedbackHistory(companyId: Long, employeeId: Long): List<FeedbackData> {
+        return selectFeedback()
+            .where(feedback.companyId.eq(companyId))
```

Comment:
filtering by companyId also repeats itself

Reply:
Got it fixed!